docs/src/testing/running-tests.md

```diff
@@ -59,3 +59,69 @@ Failure data:
 [SKIP] package_name::yet_another_test
 Tests: 3 passed, 1 failed, 2 skipped
 ```
+
+## Running tests in Scarb Workspace
+
+When running `snforge` in a Scarb workspace with a root package, it will only run tests inside the root package. When it's ran in a virtual workspace it will execute tests for all packages.  
+
+For a project structure like this
+
+```shell
+$ tree . -L 3
+.
+├── Scarb.toml
+├── crates
+│   ├── addition
+│   │   ├── Scarb.toml
+│   │   ├── src
+│   │   └── tests
+│   └── fibonacci
+│       ├── Scarb.toml
+│       └── src
+└── src
+    └── lib.cairo
+$ snforge
+Collected 1 test(s) and 1 test file(s)
+Running 1 test(s) from hello_workspaces package
+[PASS] hello_workspaces::test_simple
+Tests: 1 passed, 0 failed, 0 skipped
+```
+
+To specify a package to test, pass a `--package package_name` (or `-p package_name` for short) flag, to select the specific package. You can also run `snforge` from the package directory for the same effect.
+
+```shell
+$ snforge --package addition
+Collected 4 test(s) and 3 test file(s)
+Running 1 test(s) from addition package
+[PASS] addition::tests::it_works
+Running 2 test(s) from tests/nested/test_nested.cairo
+[PASS] test_nested::test_two
+[PASS] test_nested::test_two_and_two
+Running 1 test(s) from tests/test_simple.cairo
+[PASS] test_simple::simple_case
+Tests: 4 passed, 0 failed, 0 skipped
+```
+
+You can also pass `--workspace` flag to explicitly run tests for all packages in the workspace.
+
+```shell
+$ snforge --workspace
+Collected 4 test(s) and 3 test file(s)
+Running 1 test(s) from addition package
+[PASS] addition::tests::it_works
+Running 2 test(s) from tests/nested/test_nested.cairo
+[PASS] test_nested::test_two
+[PASS] test_nested::test_two_and_two
+Running 1 test(s) from tests/test_simple.cairo
+[PASS] test_simple::simple_case
+Tests: 4 passed, 0 failed, 0 skipped
+Running 1 test(s) from fibonacci package
+[PASS] fibonacci::tests::it_works
+Tests: 1 passed, 0 failed, 0 skipped
+Collected 3 test(s) and 2 test file(s)
+Running 1 test(s) from hello_workspaces package
+[PASS] hello_workspaces::test_simple
+Tests: 1 passed, 0 failed, 0 skipped
+```
```

Comment:
I think a separate subsection showcasing how scarb behaves in virtual workspaces could be useful as well.

Reply:
Yeah, sorry. I meant `snforge`.